### PR TITLE
#SZkg6SnD Setting up multiple languages

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
**What does this PR do?**
This PR sets up multiple languages.

**Description of Task to be completed?**
The users should be able to change/select a language of their choice.
The site contents are presented in the newly set language.
At least, two languages are supported.

**How should this be manually and online tested?**
Clone the repo
run `npm install` in your terminal
run `npm run index` in your terminal
run `http:localhost:3020` in your postman/browser
run `http:localhost:3020/?lang=fr` in your postman/browser
run` http:localhost:3020/?lang=kiny` in your postman/browser

**What are the relevant Trello id stories?**
https://trello.com/c/SZkg6SnD/20-support-multiple-languages

**Screenshots**
image
image
image

### CODE REVIEW

**Review by Mugema123 — APPROVED** [review]
After running all tests of different languages, it is working properly.

**Comment by Ntwali-Josue** [comment]
Thanks, @Mugema123!

**Comment by Ally4** [comment]
Update the PR adding the installation of i18n, otherwise, it works!.

File: package.json [hunk 1]
```diff
@@ -27,11 +27,16 @@
...
     "@babel/runtime": "^7.11.2",
+    "awilix": "^4.2.6",
+    "consolidate": "^0.16.0",
+    "cookie-parser": "^1.4.5",
     "cors": "^2.8.5",
     "cross-env": "^7.0.2",
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
     "glob": "^7.1.6",
+    "i18n": "^0.13.2",
```
Comment (Ntwali-Josue) [inline package.json:37]: @Ally4 I've installed and added the i18n you recently asked on.

**Review by Ally4 — APPROVED** [review]

**Comment by Ally4** [comment]
what if you could write a script in package.json to run `node --experimental-modules index.mjs`

**Review by TuyizeeAnastase — APPROVED** [review]
Good. when see French pop give option to in English, an i can Kinyarwanda. Cool all task reviewed as done.

**Comment by Ntwali-Josue** [comment]
@Ally4 I'm going to work on that

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+<<<<<<< HEAD
 /node_modules
 /.nyc_output
 /coverage
+=======
+node_modules
+>>>>>>> c2143c8... chore(setup): setting up node.js
+=======
+node_modules
+>>>>>>> cb94787... chore(setup): setting up node.js
```
Comment (blaise82) [inline .gitignore]: make sure that you remove the conflicts

File: index.mjs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable import/no-named-as-default */
+/* eslint-disable import/no-named-as-default-member */
+/* eslint-disable no-console */
+// eslint-disable-next-line import/extensions
+/* eslint-disable import/extensions */
+/* eslint-disable no-console */
+/* eslint-disable no-console */
+// eslint-disable-next-line import/extensions
```
Comment (blaise82) [inline index.mjs]: don't be disabling any rules

File: index.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable import/no-named-as-default */
+/* eslint-disable import/no-named-as-default-member */
+/* eslint-disable no-console */
+// eslint-disable-next-line import/extensions
+/* eslint-disable import/extensions */
+/* eslint-disable no-console */
+/* eslint-disable no-console */
+// eslint-disable-next-line import/extensions
```
Comment (blaise82) [inline index.js]: don't ignore any eslint rule
and all source file must be inside the server folder so this shouldn't be here

File: languages/server.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable linebreak-style */
+
+import express from 'express';
+import consolidate from 'consolidate';
+import cookieParser from 'cookie-parser';
+
+export class Server {
+  constructor({ i18nProvider, localeService }) {
+    this.express = express();
```
Comment (blaise82) [inline languages/server.js]: so language has it's own server?

**Review by blaise82 — CHANGES_REQUESTED** [review]
rename the server folder as src folder
improve your folder structure (remove views, remove any other file or folder outside the src folder)

File: languages/server.js (hunk 5 above)
Comment (Ntwali-Josue) [reply languages/server.js]: I've changed it,we're now having the same server

File: .gitignore (hunk 2 above)
Comment (Ntwali-Josue) [reply .gitignore]: all conflicts are solved

File: index.mjs (hunk 3 above)
Comment (Ntwali-Josue) [reply index.mjs]: Undisabling rules solved

File: index.js (hunk 4 above)
Comment (Ntwali-Josue) [reply index.js]: All source files are now in src folder

**Review by blaise82 — APPROVED** [review]

**Pull request merged**